rhel/vex/parser.go

```diff
@@ -501,11 +546,41 @@ func cvssVectorFromScore(sc *csaf.Score) (vec string, err error) {
 
 // CreatePackageKey creates a unique key to describe an arch agnostic
 // package for deduplication purposes.
-// i.e. AppStream-8.2.0.Z.TUS:python3-idle-0:3.6.8-24.el8_2.2
-func createPackageKey(repo, name, fixedIn string) string {
+// i.e. AppStream-8.2.0.Z.TUS:a_module:python3-idle-0:3.6.8-24.el8_2.2
+func createPackageKey(repo, mod, name, fixedIn string) string {
 	// The other option here is just to use repo + PURL string
 	// w/o the qualifiers I suppose instead of repo + NEVR.
-	return repo + ":" + name + "-" + fixedIn
+	return repo + ":" + mod + ":" + name + "-" + fixedIn
+}
+
+func createPackageModule(p *csaf.Product) (string, error) {
+	var modName string
+	modPURLHelper, ok := p.IdentificationHelper["purl"]
+	if !ok {
+		return modName, nil
+	}
+	purl, err := packageurl.FromString(modPURLHelper)
+	if err != nil {
+		return "", err
+	}
+	if purl.Type != "rpmmod" {
+		return "", fmt.Errorf("invalid RPM module PURL: %q", purl.String())
+	}
+	switch {
+	case purl.Namespace == "redhat":
+		if version, _, found := strings.Cut(purl.Version, ":"); found {
+			modName = purl.Name + ":" + version
+		} else {
+			modName = purl.Name + ":" + purl.Version
+		}
+	case strings.HasPrefix(purl.Namespace, "redhat") && strings.Contains(purl.Namespace, "/"):
+		// Account for the case where the PURL is unconventional
+		// e.g. pkg:rpmmod/redhat/postgresql:15/postgresql
+		if _, module, found := strings.Cut(purl.Namespace, "/"); found {
```

Comment:
nit: up to you, but we know at this point `found == true` because of the condition above, so we can just use do something like

```
_, modName, _ = strings.Cut(purl.Namespace, "/")
```

Reply:
Yeah, I think this is fair